src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,166 @@
+<template>
+  <VButton
+    v-show="showButton"
+    :variant="button.variant"
+    class="self-center"
+    :pressed="pressed"
+    aria-controls="filter-sidebar"
+    @click="toggleFilters"
+  >
+    <VIcon
+      v-show="icon.show"
+      :icon-path="filterIcon"
+      :class="{ 'me-2': icon.needsPadding }"
+    />
+    <span class="filter-label" :class="button.a11yProps">{{
+      button.label
+    }}</span>
+  </VButton>
+</template>
+
+<script>
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  reactive,
+  useContext,
+  watchEffect,
+} from '@nuxtjs/composition-api'
+import { isScreen } from '~/composables/use-media-query'
+import { useFilterSidebarVisibility } from '~/composables/use-filter-sidebar-visibility'
+
+import filterIcon from '~/assets/icons/filter.svg'
+import VButton from '~/components/VButton.vue'
+import VIcon from '~/components/VIcon/VIcon'
+
+const VFilterButton = defineComponent({
+  name: 'VFilterButton',
+  components: {
+    VIcon,
+    VButton,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true:
+     * - button has no border.
+     * - on mobile, the button label is blank.
+
+     * @default false
+     */
+    isHeaderScrolled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const { i18n, store } = useContext()
+    const isMdScreen = isScreen('md')
+
+    const {
+      isFilterSidebarVisible: pressed,
+      setFilterSidebarVisibility,
+    } = useFilterSidebarVisibility({ mediaQuery: isMdScreen })
+
+    const appliedFilterTags = computed(
+      () => store.getters['search/appliedFilterTags'].length
+    )
+    const isAnyFilterApplied = computed(() => appliedFilterTags.value > 0)
+
+    const labelWithCount = (count) => {
+      return i18n.tc('header.filter-button.with-count', count)
+    }
+    // TODO(obulat): Remove buttonVariants. The aim of this object was to lower
+    // the cognitive load of remembering the variant names when debugging the
+    // button.
+    const buttonVariants = {
+      pressed: 'action-menu-pressed',
```

Comment:
I guess this is _technically_ a valid variant that you can pass to the button component in that it will render correctly... but really it should be a combination of the `action-menu` variant and `:pressed="true"` prop. Ideally `pressed` would just be passed through to the button component and all of the pressed styles for the variants used by this component would be the same?

I'm still kind of sad that we need three _slightly_ different variants for this component 😞

Reply:
I didn't read the code well-enough to notice that there is no 'action-menu-pressed' variant, sorry about that. I removed the 'pressed' variant, and only left the two different 'action-menu' variants: one with a border, and one without. I agree that it should be possible to join these two variants in favor of additional prop? or some other config?

Reply:
Maybe we need a `pressed` and `active` prop? 🤔 For most variants this is the same state I think and I'd have a hard time understanding the difference between those two props at face value personally... hmmm. Gutenberg has an `is-busy` prop but that adds an animation, and I don't think it quite fits with our situation here.

We could just remove the `muted` variant and add that as a custom state to the parent VFilterButton instead.